pkg/datasources/datasource_network_policy.go

```diff
@@ -0,0 +1,120 @@
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The network policies in the account",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "The ID of the network policy.",
+							Computed:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name of the network policy.",
+							Computed:    true,
+						},
+						"comment": {
+							Type:        schema.TypeString,
+							Description: "The comment of the network policy.",
+							Computed:    true,
+						},
+						"rules": {
+							Type:        schema.TypeList,
+							Description: "Rules for the network policy.",
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:        schema.TypeString,
+										Description: "The name of the rule.",
+										Computed:    true,
+									},
+									"action": {
+										Type:        schema.TypeString,
+										Description: "The action to take for this rule. Currently only 'allow' is supported.",
+										Computed:    true,
+									},
+									"direction": {
+										Type:        schema.TypeString,
+										Description: "The direction of traffic the rule applies to. Currently only 'ingress' is supported.",
+										Computed:    true,
+									},
+									"address": {
+										Type:        schema.TypeString,
+										Description: "The CIDR block the rule will be applied to.",
+										Computed:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"region": RegionSchema(),
+		},
+	}
+}
+
+func networkPolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Add List function to materialize package
+	networkPolicy, err := materialize.ListNetworkPolicies(metaDb)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	networkPolicyFormatted := []map[string]interface{}{}
+	for _, p := range networkPolicy {
+		networkPolicyMap := map[string]interface{}{}
+
+		networkPolicyMap["id"] = p.PolicyId.String
+		networkPolicyMap["name"] = p.PolicyName.String
+		networkPolicyMap["comment"] = p.Comment.String
+
+		// Format rules
+		rules := []map[string]interface{}{}
+		for _, r := range p.Rules {
+			ruleMap := map[string]interface{}{
+				"name":      r.Name,
+				"action":    r.Action,
+				"direction": r.Direction,
+				"address":   r.Address,
+			}
+			rules = append(rules, ruleMap)
+		}
+		networkPolicyMap["rules"] = rules
+
+		networkPolicyFormatted = append(networkPolicyFormatted, networkPolicyMap)
+	}
+
+	if err := d.Set("network_policies", networkPolicyFormatted); err != nil {
+		return diag.FromErr(err)
+	}
+
+	SetId(string(region), "network_policies", "", "", d)
```

Comment:
nit: Noticed we use d.SetId in some places and SetId in other places. Any reason why we don't use `d.SetId(transformIdWithRegion...` here?